### PR TITLE
chore(updatecli) improve trusted.ci credential manifests (documentation, native HCL resource)

### PR DESCRIPTION
**Pull request by dduportal**
Example results (with dummy dates):

- https://github.com/jenkins-infra/azure/pull/834
- https://github.com/jenkins-infra/azure/pull/835
- https://github.com/jenkins-infra/azure/pull/836
- https://github.com/jenkins-infra/azure/pull/837

### CODE REVIEW

File: updatecli/values.yaml [hunk 1]
```diff
@@ -32,18 +32,42 @@ end_dates:
...
+        > ⚠️ Make sure you can generate this ZIP file BEFORE merging the pull request here!
     trustedci_updatesjenkinsio_redirections_fileshare_serviceprincipal_writer:
-      end_date: 2024-12-18T00:00:00Z
       service: "updates.jenkins.io (redirections)"
-      secret: "TODO: to be identified, related to updates.jenkins.io (redirections)"
+      doc_how_to_get_credential: |
+        > [!IMPORTANT]
+        >
+        > ⚠️ Merging this PR will prevent the Update Center job (update_center2) to succeed and update the Update Center index.
+        > You'll have to update the top-level credential `update-center-publish-env` on trusted.ci.jenkins.io.
```
Comment (dduportal) [inline updatecli/values.yaml:51]:
```suggestion
        > You'll have to update the top-level credential `update-center-publish-env` on trusted.ci.jenkins.io.
```
```suggestion
        > You'll have to update the top-level credential `update-center-publish-env` on trusted.ci.jenkins.io UI.
```

File: updatecli/values.yaml [hunk 2]
```diff
@@ -32,18 +32,42 @@ end_dates:
...
   trusted_ci_jenkins_io:
     trustedci_updatesjenkinsio_content_fileshare_serviceprincipal_writer:
-      end_date: 2024-12-18T00:00:00Z
       service: "updates.jenkins.io (content)"
-      secret: "trusted_ci_jenkins_io_fileshare_serviceprincipal_writer"
+      doc_how_to_get_credential: |
+        > [!IMPORTANT]
+        >
+        > ⚠️ Merging this PR will prevent the Update Center job (update_center2) to succeed and update the Update Center index.
+        > You'll have to update the top-level credential `update-center-publish-env` on trusted.ci.jenkins.io with the ZIP file.
```
Comment (dduportal) [inline updatecli/values.yaml:40]:
```suggestion
        > You'll have to update the top-level credential `update-center-publish-env` on trusted.ci.jenkins.io with the ZIP file.
```
```suggestion
        > You'll have to update the top-level credential `update-center-publish-env` on trusted.ci.jenkins.io UI with the ZIP file.
```

File: updatecli/values.yaml [hunk 3]
```diff
@@ -32,18 +32,42 @@ end_dates:
...
+        > This credential value can be retrieved in the Terraform state from "module.trustedci_jenkinsio_fileshare_serviceprincipal_writer.fileshare_serviceprincipal_writer_application_client_password".
     trustedci_javadocjenkinsio_fileshare_serviceprincipal_writer:
-      end_date: 2024-10-16T00:00:00Z
       service: "javadoc.jenkins.io"
-      secret: "trustedci_javadocjenkinsio_fileshare_serviceprincipal_writer"
+      doc_how_to_get_credential: |
+        > [!IMPORTANT]
+        >
+        > ⚠️ Merging this PR will prevent the "jenkins.io" deployment job to succeed and update the www.jenkins.io website.
+        > You'll have to update the top-level credential `trustedci_javadocjenkinsio_fileshare_serviceprincipal_writer` on trusted.ci.jenkins.io.
```
Comment (dduportal) [inline updatecli/values.yaml:71]:
```suggestion
        > You'll have to update the top-level credential `trustedci_javadocjenkinsio_fileshare_serviceprincipal_writer` on trusted.ci.jenkins.io.
```
```suggestion
        > You'll have to update the top-level credential `trustedci_javadocjenkinsio_fileshare_serviceprincipal_writer` on trusted.ci.jenkins.io UI.
```

File: updatecli/values.yaml [hunk 4]
```diff
@@ -32,18 +32,42 @@ end_dates:
       secret: "STATS_SERVICE_PRINCIPAL_WRITER_CLIENT_SECRET"
   trusted_ci_jenkins_io:
     trustedci_updatesjenkinsio_content_fileshare_serviceprincipal_writer:
-      end_date: 2024-12-18T00:00:00Z
       service: "updates.jenkins.io (content)"
-      secret: "trusted_ci_jenkins_io_fileshare_serviceprincipal_writer"
+      doc_how_to_get_credential: |
+        > [!IMPORTANT]
+        >
+        > ⚠️ Merging this PR will prevent the Update Center job (update_center2) to succeed and update the Update Center index.
```
Comment (smerle33) [inline updatecli/values.yaml:39]:
```suggestion
        > ⚠️ Merging this PR will prevent the Update Center job (update_center2) to succeed and update the Update Center index.
        >put the update_center job in disable mode during the time between the merge and the credential upload
```

File: updatecli/values.yaml [hunk 5]
```diff
@@ -32,18 +32,42 @@ end_dates:
...
-      secret: "TODO: to be identified, related to updates.jenkins.io (redirections)"
+      doc_how_to_get_credential: |
+        > [!IMPORTANT]
+        >
+        > ⚠️ Merging this PR will prevent the Update Center job (update_center2) to succeed and update the Update Center index.
+        > You'll have to update the top-level credential `update-center-publish-env` on trusted.ci.jenkins.io.
+        >
+        > This credential is a ZIP file generated by the (private) script:
+        https://github.com/jenkins-infra/charts-secrets/blob/main/config/trusted.ci.jenkins.io/get-uc-sync-zip-credential.sh.
+        > ⚠️ Make sure you can generate this ZIP file BEFORE merging the pull request here!
```
Comment (smerle33) [inline updatecli/values.yaml:55]:
```suggestion
        > ⚠️ Make sure you can generate this ZIP file BEFORE merging the pull request here!
        > ⚠️ Check if you don't have other PR to deal with on that specific zip credential, you could then merge both and run only once the generator script
```

File: updatecli/values.yaml [hunk 6]
```diff
@@ -32,18 +32,44 @@ end_dates:
...
+        > ⚠️ Check if you don't have other PR to related to the same ZIP credential to only generates the ZIP once.
     trustedci_updatesjenkinsio_redirections_fileshare_serviceprincipal_writer:
-      end_date: 2024-12-18T00:00:00Z
       service: "updates.jenkins.io (redirections)"
-      secret: "TODO: to be identified, related to updates.jenkins.io (redirections)"
+      doc_how_to_get_credential: |
+        > [!IMPORTANT]
+        >
+        > ⚠️ Merging this PR will prevent the Update Center job (update_center2) to succeed and update the Update Center index.
```
Comment (smerle33) [inline updatecli/values.yaml:51]:
```suggestion
        > ⚠️ Merging this PR will prevent the Update Center job (update_center2) to succeed and update the Update Center index. You should disable the `update_center2` job before merging, between the PR merge and the credential upload.
```

**Comment by dduportal** [comment]
Closes #834
Closes #835
Closes #836
Closes #837

**Pull request merged**